### PR TITLE
typing: run pyright on examples

### PR DESCRIPTION
**Pull request by shiftinv**
## Summary

Adds the `examples/` directory to the pyright includes, and fixes several minor pyright errors in the examples that appeared as a result.

depends on #563

## Checklist

- [x] If code changes were made, then they have been tested
    - [ ] I have updated the documentation to reflect the changes
    - [x] I have formatted the code properly by running `task lint`
    - [x] I have type-checked the code by running `task pyright`
- [x] This PR fixes an issue
- [ ] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: examples/user_echo_webhook.py [hunk 1]
```diff
@@ -6,9 +6,8 @@
                  bot will create a webhook to "impersonate"
                  specified user, so to speak."""
 
-intents = (
-    disnake.Intents.all()
-)  # Not limiting the permissions, since this is an example - set this accordingly.
+# Not limiting the permissions, since this is an example - set this accordingly.
```
Comment (onerandomusername) [inline examples/user_echo_webhook.py:9]: intents

Comment (shiftinv) [reply examples/user_echo_webhook.py]: fixed in 89a5c65121de57610b8116404aa94c288f5e53b0

File: examples/user_echo_webhook.py [hunk 2]
```diff
@@ -31,23 +30,25 @@ async def userecho(ctx, member: disnake.Member, *, content):
...
     channel_webhooks = await ctx.message.channel.webhooks()
-    webhook_count = 0
 
     # We check if the bot's webhook already exists in the channel.
     for webhook in channel_webhooks:
         # We will check if the creator of the webhook is the same as the bot, and if the name is the same.
-        if webhook.user.id == client.user.id and webhook.name == "Bot Webhook":
+        if webhook.user == client.user and webhook.name == "Bot Webhook":
             await webhook.send(
                 content=content, username=member.display_name, avatar_url=member.avatar
```
[the remark this answers is not in the document]
Comment (shiftinv) [reply examples/user_echo_webhook.py]: refactored the example further, change not required anymore as of c619d114f7b5d79118c1918480ab82ae0c661b2b

**Comment by onerandomusername** [comment]
@shiftinv hi, could you please resolve conflicts?

**Comment by shiftinv** [comment]
> @shiftinv hi, could you please resolve conflicts?

Done, though one example currently doesn't type-check correctly likely due to a change in #462. Will try to figure that one out first 🤔

**Comment by onerandomusername** [comment]
That issue that was caused by #462 looks like a bug in 462.

As far as I can tell, we need to replace all occurrences of ``V = TypeVar("V", "View", None, covariant=True)`` with ``V = TypeVar("V", bound=Optional["View"], covariant=True)``.

**Comment by Chromosomologist** [comment]
> As far as I can tell, we need to replace all occurrences of V = TypeVar("V", "View", None, covariant=True) with V = TypeVar("V", bound=Optional["View"], covariant=True)
Yup, that seems to retain the intended behaviour where `Button()` return a `Button[None]` and also fix this issue with inheritance.

**Comment by shiftinv** [comment]
Moved all the typing-related lib changes into #563 and only kept the ones specific to examples in this PR, as originally intended. As such, this now depends on #563 for checks to pass

**Review by onerandomusername — APPROVED** [review]
this looks good, however I did not test the changed examples due to some issues that are well out of the scope of this pull. 

This pr does as it says, and does not change the above discovered issues any more or less.

**Pull request merged**